sonic_platform_base/sonic_xcvr/api/public/cmisCDB.py

```diff
@@ -0,0 +1,430 @@
+        else:
+            return True
+    
+    def cdb_chkcode(self, cmd):
+        '''
+        This function calculates and returns the checksum of a CDB command
+        '''
+        checksum = 0
+        for byte in cmd:
+            checksum += byte   
+        return 0xff - (checksum & 0xff)
+
+    def cdb1_chkstatus(self):
+        '''
+        This function checks the CDB status.
+        The format of returned values is busy flag, failed flag and cause
+
+        CDB command status
+        Bit 7: CdbIsBusy
+        Bit 6: CdbHasFailed
+        Bit 5-0: CdBCommandResult
+        Coarse Status     CdbIsBusy       CdbHasFailed
+        IN PROGRESS       1               X (dont care)
+        SUCCESS           0               0
+        FAILED            0               1
+
+        IN PROGRESS
+            00h=Reserved
+            01h=Command is captured but not processed
+            02h=Command checking is in progress
+            03h=Previous CMD was ABORTED by CMD Abort
+            04h-1Fh=Reserved
+            20h-2Fh=Reserved
+            30h-3Fh=Custom
+
+        SUCCESS
+            00h=Reserved
+            01h=Command completed successfully
+            02h=Reserved
+            03h=Previous CMD was ABORTED by CMD Abort
+            04h-1Fh=Reserved
+            20h-2Fh=Reserved
+            30h-3Fh=Custom
+
+        FAILED
+            00h=Reserved
+            01h=CMDCode unknown
+            02h=Parameter range error or parameter not supported
+            03h=Previous CMD was not ABORTED by CMD Abort
+            04h=Command checking time out
+            05h=CdbCheckCode Error
+            06h=Password related error (command specific meaning)
+            07h=Command not compatible with operating status
+            08h-0Fh=Reserved for STS command checking error
+            10h-1Fh=Reserved
+            20h-2Fh=For individual STS command or task error
+            30h-3Fh=Custom
+        '''
+        status = self.xcvr_eeprom.read(consts.CDB1_STATUS)
+        is_busy =  bool((status >> 7) & 0x1)
+        cnt = 0
+        while is_busy and cnt < MAX_WAIT:
+            time.sleep(1)
+            is_busy =  bool((status >> 7) & 0x1)
+            cnt += 1
+        return status
+
+    def write_cdb(self, cmd):
+        '''
+        This function writes a CDB command to page 0x9f
+        '''
+        self.xcvr_eeprom.write_raw(LPLPAGE*PAGE_LENGTH+CDB_WRITE_MSG_START, len(cmd)-CMDLEN, cmd[CMDLEN:])
+        self.xcvr_eeprom.write_raw(LPLPAGE*PAGE_LENGTH+INIT_OFFSET, CMDLEN, cmd[:CMDLEN])
+
+    def read_cdb(self):
+        '''
+        This function reads the reply of a CDB command from page 0x9f
+        '''
+        rpllen = self.xcvr_eeprom.read(consts.CDB_RPL_LENGTH)
+        rpl_chkcode = self.xcvr_eeprom.read(consts.CDB_RPL_CHKCODE)
+        rpl = self.xcvr_eeprom.read_raw(LPLPAGE*PAGE_LENGTH+CDB_RPL_OFFSET, rpllen)
+        return rpllen, rpl_chkcode, rpl
+
+    # Query status
+    def cmd0000h(self):
```

Comment:
please rename to meaningful function name

Reply:
done